src/JuliusSweetland.OptiKey.Core/UI/ViewModels/MainViewModel.ServiceEventHandlers.cs

```diff
@@ -2372,47 +2380,117 @@ private void SelectVoice(string voice)
             NavigateToMenu();
         }
 
-        private void StepList(string command)
+        public void StepList(KeyValue singleKeyValue, List<string> multiKeySelection)
+        {
+            if (!string.IsNullOrEmpty(singleKeyValue.String))
+            {
+                Log.InfoFormat("StepList called with singleKeyValue [{0}]", singleKeyValue.String);
+
+                List<string> commandList = singleKeyValue.String.Split('<').ToList();
+                if (commandList.Any())
+                {
+                    var cTask = Task.Run(() => StepListCommands(singleKeyValue, multiKeySelection, commandList));
+                }
+            }
+        }
+
+        private async Task<int> StepListCommands(KeyValue singleKeyValue, List<string> multiKeySelection, List<string> commandList)
```

Comment:
@JuliusSweetland I made this method into an async task.

Reply:
@AdamRoden The problem is that nothing is awaiting that returned task:

Task.Run(() => StepListCommands(singleKeyValue, multiKeySelection, commandList));

^^^ that doesn't await an async method, it merely runs it on a threadpool thread.

Reply:
You COULD do a Task.Run and then call .Wait on the StepListCommands() call, i.e. StepListCommands(parameters).Wait(), but I still think that running the step list on a background thread is not necessary. Starting a new thread "costs" about 1MB of memory (not 2 as I previously said) and you will later have to dispatch back to the UI thread anyway.

I'll email you my version of the class which is async/await all the way down to the StepList method. Can you compare and see if you can carry on the async/await pattern into the delays etc?